src/include/transaction/timestamp_manager.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+#include <algorithm>
+#include <unordered_set>
+#include "common/spin_latch.h"
+#include "common/strong_typedef.h"
+#include "transaction/transaction_defs.h"
+
+namespace terrier::transaction {
+class TransactionManager;
+/**
+ * Generates timestamps, and keeps track of the lifetime of transactions (whether they have entered or left the system)
+ */
+class TimestampManager {
```

Comment:
I’m confused as to why we needed to separate this class from the transaction manager if the majority of its code is specific to tracking transactions.

Specifically, why is it not solely the transaction manager’s job to track live transactions?

Reply:
circular dependency between DeferredActionManager and TransactionManager for DI. (DAM cannot function without a pointer to TM, TM cannot function without owning a DAM pointer.) This is the easiest way to break that cycle.

Reply:
Should we generalize the nomenclature to tracking “active timestamps“?  The dependency issue makes sense, but I think we need to make a cleaner separation between the two classes even if it’s just terminology.

Reply:
I was thinking about that, but then, this concept exists to manage transactions. Inventing another concept of "active timestamps" seems counterproductive.